File: Xcode Templates/Glyphs Dev/Glyphs Palette Plugin.xctemplate/___PACKAGENAMEASIDENTIFIER___.m
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//  ___COPYRIGHT___
//

#import "___PACKAGENAMEASIDENTIFIER___.h"
#import <GlyphsApp/GSPaletteView.h>

@implementation ___FILEBASENAMEASIDENTIFIER___

@synthesize windowController;

- (instancetype)init {
	self = [super initWithNibName:@"___PACKAGENAMEASIDENTIFIER___View" bundle:[NSBundle bundleForClass:[self class]]];
	[(GSPaletteView *)self.view setController:self];
	return self;
}

- (NSUInteger)interfaceVersion {
	// Distinguishes the API version the plugin was built for. Return 1.
	return 1;
}

- (NSString *)title {
	// Return the name of the tool as it will appear in the menu.
	return @"___PACKAGENAME___";
}

- (NSInteger)minHeight {
	return 85;
}

- (NSInteger)maxHeight {
	return 265; // if this is bigger than minHeight, the palette is resizable
}

- (NSUInteger)currentHeight {
	return [NSUserDefaults.standardUserDefaults integerForKey:@"___PACKAGENAMEASIDENTIFIER___CurrentHeight"];
}

- (void)setCurrentHeight:(NSUInteger)newHeight {
	if (newHeight >= [self minHeight] && newHeight <= [self maxHeight]) {
		[NSUserDefaults.standardUserDefaults setInteger:newHeight forKey:@"___PACKAGENAMEASIDENTIFIER___CurrentHeight"];
	}
}

- (NSView *)theView {
	return [self view];
}

@end

File: Xcode Templates/Glyphs Dev/Glyphs Palette Plugin.xctemplate/___PACKAGENAMEASIDENTIFIER___.swift
//
//  ___FILENAME___
//  ___PACKAGENAME___
//
//  Created by ___FULLUSERNAME___ on ___DATE___.
//___COPYRIGHT___
//

import Cocoa

@objc class ___FILEBASENAMEASIDENTIFIER___: NSViewController, GlyphsPalette {

	@MainActor weak var windowController: (NSWindowController & GSWindowControllerProtocol)?

	// Distinguishes the API version the plugin was built for. Return 1.
	let interfaceVersion: UInt = 1

	required init?(coder: NSCoder) {
		super.init(coder: coder)
	}

	override init(nibName nibNameOrNil: NSNib.Name?, bundle nibBundleOrNil: Bundle?) {
		super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
	}

	convenience init() {
		self.init(nibName: "___FILEBASENAMEASIDENTIFIER___View", bundle: Bundle(for: Self.self))
		(self.view as? GSPaletteView)?.controller = self
	}

	func loadPlugin() {
		self.title = "___PACKAGENAME___"
	}

	var minHeight: Int {
		return 85
	}

	var maxHeight: Int {
		return 265 // if this is bigger than minHeight, the palette is resizable
	}

	var currentHeight: UInt {
		get {
			return UInt(UserDefaults.standard.integer(forKey: "___FILEBASENAMEASIDENTIFIER___CurrentHeight"))
		}
		set {
			if newValue >= minHeight && newValue <= maxHeight {
				UserDefaults.standard.set(Int(newValue), forKey: "___FILEBASENAMEASIDENTIFIER___CurrentHeight")
			}
		}
	}

	var theView: NSView {
		return view
	}
}
